File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        zero_flipped=0
        maxi=0
        start=0
        print nums
        for i in range(len(nums)):
            if(nums[i]==0):
                zero_flipped=zero_flipped+1
            while(zero_flipped>k):
                if(nums[start]==0):
                    zero_flipped=zero_flipped-1
                start=start+1
            maxi=max({maxi,i-start+1})
        return maxi
        

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.java
import java.util.Arrays;
class Solution {
    public int longestOnes(int[] nums,int k) {
        int zeros_flipped=0,max=0,start=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                zeros_flipped++;
            }
            while(zeros_flipped>k){
                if(nums[start]==0){
                    zeros_flipped--;
                }
                start++;
            }
            max=Math.max(max,i-start+1);
        }
        return max;
    }
}
